kore/src/Kore/Step/SMT/AST.hs

```diff
@@ -0,0 +1,178 @@
+{-|
+Module      : Kore.Step.SMT.AST
+Description : Data types involved in declaring and using SMT symbols.
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+Maintainer  : virgil.serbanuta@runtimeverification.com
+-}
+
+module Kore.Step.SMT.AST
+    ( Declarations (..)
+    , Encodable (AlreadyEncoded)
+    , KoreSortDeclaration (..)
+    , KoreSymbolDeclaration (..)
+    , SmtDeclarations
+    , SmtKoreSymbolDeclaration
+    , SmtSort
+    , SmtSymbol
+    , Sort (..)
+    , SortReference (..)
+    , Symbol (..)
+    , SymbolReference (..)
+    , UnresolvedConstructor
+    , UnresolvedConstructorArgument
+    , UnresolvedDataTypeDeclaration
+    , UnresolvedDeclarations
+    , UnresolvedFunctionDeclaration
+    , UnresolvedKoreSortDeclaration
+    , UnresolvedKoreSymbolDeclaration
+    , UnresolvedSort
+    , UnresolvedSortDeclaration
+    , UnresolvedSymbol
+    , appendToEncoding
+    , encodable
+    , encode
+    , mergePreferFirst
+    ) where
+
+import qualified Data.Map.Merge.Strict as Map.Merge
+import           Data.Map.Strict
+                 ( Map )
+import           Data.Text
+                 ( Text )
+
+import qualified Kore.AST.Identifier as Kore
+                 ( Id (Id, getId) )
+import           Kore.AST.MetaOrObject
+                 ( Object )
+import qualified Kore.Sort as Kore
+                 ( Sort )
+import           Kore.Step.SMT.Encoder
+                 ( encodeName )
+import qualified SMT.AST as AST
+
+data Sort sort symbol name =
```

Comment:
Could you give a comment describing how this corresponds to an SMT sort? Especially the `smtFromSortArgs` field.

Reply:
Oops, sorry, I thought I also added comments to this file.